tests/handlers/auth_test.py

```diff
@@ -0,0 +1,358 @@
+
+
+async def test_authnz_token_access_denied(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair)
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 403
+    body = await r.text()
+    assert "No Capability group found in user's `isMemberOf`" in body
+    assert r.headers["X-Auth-Request-Token-Capabilities"] == ""
+    assert r.headers["X-Auth-Request-Capabilities-Accepted"] == "exec:admin"
+    assert r.headers["X-Auth-Request-Capabilities-Satisfy"] == "all"
+
+
+async def test_authnz_token_satisfy_all(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["test"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params=[("capability", "exec:test"), ("capability", "exec:admin")],
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 403
+    body = await r.text()
+    assert "No Capability group found in user's `isMemberOf`" in body
+    assert r.headers["X-Auth-Request-Token-Capabilities"] == "exec:test"
+    assert (
+        r.headers["X-Auth-Request-Capabilities-Accepted"]
+        == "exec:admin exec:test"
+    )
+    assert r.headers["X-Auth-Request-Capabilities-Satisfy"] == "all"
+
+
+async def test_authnz_token_success(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["admin"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={"Authorization": f"Bearer {token}"},
+    )
+    assert r.status == 200
+    assert (
+        r.headers["X-Auth-Request-Token-Capabilities"] == "exec:admin read:all"
+    )
+    assert r.headers["X-Auth-Request-Capabilities-Accepted"] == "exec:admin"
+    assert r.headers["X-Auth-Request-Capabilities-Satisfy"] == "all"
+    assert r.headers["X-Auth-Request-Email"] == "some-user@example.com"
+    assert r.headers["X-Auth-Request-User"] == "some-user"
+    assert r.headers["X-Auth-Request-Uid"] == "1000"
+    assert r.headers["X-Auth-Request-Groups"] == "admin"
+    assert r.headers["X-Auth-Request-Token"] == token
+    assert r.headers["X-Auth-Request-Token-Ticket"] == ""
+
+
+async def test_authnz_token_success_any(aiohttp_client: TestClient) -> None:
+    keypair = RSAKeyPair()
+    token = create_test_token(keypair, ["test"])
+    app = await create_test_app(keypair)
+    client = await aiohttp_client(app)
+
+    r = await client.get(
+        "/auth",
+        params=[
+            ("capability", "exec:admin"),
+            ("capability", "exec:test"),
+            ("satisfy", "any"),
```

Comment:
I'm a bit confused by this test... satisfy any, and then the capabilities on :151 doesn't have the admin in there?  Is that the capability it's picking to satisfy something?

Reply:
The idea is that either `exec:admin` or `exec:test` is sufficient, so as a test we ask for both with satisfy `any` and then pass in credentials that only have `exec:test` and make sure they're accepted but also don't erroneously claim we have `exec:admin`.